docs/transformation/jsontransformation.md

```diff
@@ -128,6 +128,27 @@ spec:
       value: ce-$source-$id
 ```
 
+**Conditional Add operation**
+
+Conditionally add a new value depending on whether or not a stored variable is instantiated.
+
+```yaml
+spec:
+  data:
+  - operation: store
+    paths:
+      - key: $name
+        value: user.name
+  - operation: add:
+    paths:
+      - key: message
+        value: (hello $name)  
+```
+
+In this example, the value of `message` will either be `hello <name>` or an `empty string` if `$name` is not set.
```

Comment:
nitpicking: it feels like anything in `this formatting` suggests that it is a literal value, i.e. resulting event will assumingely contain `message: empty string`, while in fact it will be just empty string, `message: ""`.

```suggestion
In this example, the value of `message` will either be `hello <name>` or an empty string if `$name` is not set.
```